tools/mtouch/Application.cs

```diff
@@ -1147,7 +1147,20 @@ void BuildFinalExecutable ()
 			} else {
 				if (!IsWatchExtension) {
 					// In extensions we need to save a list of the frameworks we need so that the main app can get them.
-					var all_frameworks = Frameworks.Union (WeakFrameworks);
+					var all_frameworks = new HashSet<string> ();
+					all_frameworks.UnionWith (Frameworks);
```

Comment:
can't the `HashSet` be build from `Frameworks` and avoid the `UnionWith` checks (if already present) ?